url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParser.java

```diff
@@ -0,0 +1,3669 @@
+               {if (true) return op.toString();}
+      break;
+    case 98:
+      op = jj_consume_token(98);
+               {if (true) return op.toString();}
+      break;
+    case 99:
+      op = jj_consume_token(99);
+               {if (true) return op.toString();}
+      break;
+    case 100:
+      op = jj_consume_token(100);
+               {if (true) return op.toString();}
+      break;
+    default:
+      jj_la1[30] = jj_gen;
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+    throw new Error("Missing return statement in function");
+  }
+
+// TableReference ::= OracleObjectName[.OracleObjectName]
+  static final public String TableReference() throws ParseException {
+  StringBuffer buf = new StringBuffer();
+  String s;
+    s = OracleObjectName();
+                             buf.append(s);
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case 93:
+      jj_consume_token(93);
+      s = OracleObjectName();
+                                  buf.append("." + s);
+      break;
+    default:
+      jj_la1[31] = jj_gen;
+      ;
+    }
+      {if (true) return buf.toString();}
+    throw new Error("Missing return statement in function");
+  }
+
+  static final public void NumOrID() throws ParseException {
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case S_IDENTIFIER:
+      jj_consume_token(S_IDENTIFIER);
+      break;
+    case S_NUMBER:
+    case 101:
+    case 102:
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case 101:
+      case 102:
+        switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+        case 101:
+          jj_consume_token(101);
+          break;
+        case 102:
+          jj_consume_token(102);
+          break;
+        default:
+          jj_la1[32] = jj_gen;
+          jj_consume_token(-1);
+          throw new ParseException();
+        }
+        break;
+      default:
+        jj_la1[33] = jj_gen;
+        ;
+      }
+      jj_consume_token(S_NUMBER);
+      break;
+    default:
+      jj_la1[34] = jj_gen;
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+  }
+
+/* --------------- General Productions ends here --------------- */
+
+/* ----------- SQL productions start here ----------------- */
+  static final public MQuery SelectStatement() throws ParseException {
+  MQuery q;
+  Vector v;
+    q = SelectWithoutOrder();
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case K_ORDER:
+      v = OrderByClause();
+                          q.addOrderBy(v);
+      break;
+    default:
+      jj_la1[35] = jj_gen;
+      ;
+    }
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case K_FOR:
+      ForUpdateClause();
+                        q.forupdate_ = true;
+      break;
+    default:
+      jj_la1[36] = jj_gen;
+      ;
+    }
+    {if (true) return q;}
+    throw new Error("Missing return statement in function");
+  }
+
+  @SuppressWarnings({ "unused", "rawtypes" })
+static final public MQuery SelectWithoutOrder() throws ParseException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 823-873 into a separate method.